src/modules/constructRelationalGraph.js

```diff
@@ -0,0 +1,265 @@
+// @flow
+import {original} from 'immer';
+import Doctor from '../resources/Doctor.json';
+import Patient from '../resources/Patient.json';
+import Department from '../resources/Department.json';
+
+export type Expression = {[key: string]: any};
+
+export type JoinCondition = {[key: string]: any};
+
+export type SelectionCondition = {[key: string]: any};
```

Comment:
Defining types for these is great. I would suggest moving them to a new `types.js` file since they aren't specific to this code. Really, it probably would have been a good idea if I had defined something like this sooner since these could be used elsewhere in the code.